### ISSUE TITLE
Pod can not access itself from a cluster IP

### ISSUE THREAD

**Issue by oilbeater** [alauda/kube-ovn#136]
Now when pod use cluster ip to access it self, the packet will be dropped. We need to trace where the packet is dropped and modify the flow to let the packet hit itself

**Comment by oilbeater** [comment]
when pod access itself form svc, vswitchd will generate theses logs
```
2019-11-20T08:52:26.175Z|00992|ofproto_dpif_xlate(handler1)|WARN|received packet on unknown port 0 on bridge br-int while processing recirc_id=0x614ff,ct_state=new|trk,ct_zone=14,ct_nw_src=10.16.0.22,ct_nw_dst=10.16.0.22,ct_nw_proto=6,ct_tp_src=34790,ct_tp_dst=80,eth,tcp,in_port=733,vlan_tci=0x0000,dl_src=00:00:00:29:83:03,dl_dst=0a:00:00:10:00:17,nw_src=10.16.0.22,nw_dst=10.16.0.22,nw_tos=0,nw_ecn=0,nw_ttl=63,tp_src=34790,tp_dst=80,tcp_flags=syn
2019-11-20T08:52:26.175Z|00993|dpif(handler1)|WARN|system@ovs-system: execute ct(commit,zone=14,label=0/0x1),11 failed (Invalid argument) on packet tcp,vlan_tci=0x0000,dl_src=00:00:00:29:83:03,dl_dst=0a:00:00:10:00:17,nw_src=10.16.0.22,nw_dst=10.16.0.22,nw_tos=0,nw_ecn=0,nw_ttl=63,tp_src=34790,tp_dst=80,tcp_flags=syn tcp_csum:84c6
 with metadata skb_priority(0),skb_mark(0),ct_state(0x21),ct_zone(0xe),ct_tuple4(src=10.16.0.22,dst=10.16.0.22,proto=6,tp_src=34790,tp_dst=80),in_port(11) mtu 0
```

Seems like that ct commit failed because the dl_src and dl_dst are not equal

**Comment by oilbeater** [comment]
One workaround is to add iptables rules in pod network to redirect access to vip to it self,  this can avoid packet going into ovs to do the ct stuffs.

steps to verify
```bash
sysctl -w net.ipv4.conf.eth0.route_localnet=1
ip addr add <svc ip> dev eth0
iptables -t nat -I PREROUTING -p <svc protocol> -d <svc ip> --dport <svc port> -j DNAT --to-destination <pod ip>:<pod port>

```

**Comment by oilbeater** [comment]
https://patchwork.ozlabs.org/project/openvswitch/list/?series=153591
we should try this ovn patch to check if the issue still exists

**Comment by oilbeater** [comment]
> https://patchwork.ozlabs.org/project/openvswitch/list/?series=153591
> we should try this ovn patch to check if the issue still exists

This path based on the new ovn that split from ovs and the code structure changed too much. For stability reason we leave this fix to the release after 1.0
